### PR TITLE
fix: Fix docker base image for Java 21

### LINKED ISSUE #2281: The Gradle plugin generates invalid Dockerfile when choosing Java 21
### Expected Behavior

If the Gradle build set Java version 21 like

```kotlin
java {
    sourceCompatibility = JavaVersion.toVersion("21")
    targetCompatibility = JavaVersion.toVersion("21")
}
```

the plugin should use the compatible base images for `Dockerfile` and `DockerfileNative`

### Description

When creating a Micronaut application that uses Java 21 

```sh
mn create-app --build=gradle_kotlin --jdk=21 --lang=java --test=junit com.example.demo
```

and the run `./gradlew dockerfile` or `./gradlew dockerfileNative`, Micronaut generates Dockerfile based on a Java 17 runtime. The native image cannot be built and the JAR based Docker image fails to start.

#### DockerfileNative

```Dockerfile
FROM ghcr.io/graalvm/native-image-community:17-ol9 AS graalvm
...
```
#### Dockerfile

```Dockerfile
FROM eclipse-temurin:17-jre-focal
...
```

#### Workaround

As a workaround for the JAR based Dockerfile I can define a different base image such as

```kotlin
tasks.named<MicronautDockerfile>("dockerfile") {
    baseImage = "azul/zulu-openjdk:21-jre"
}
```

For the DockerfileNative 

```kotlin
tasks.named<io.micronaut.gradle.docker.NativeImageDockerfile>("dockerfileNative") {
    graalImage = "ghcr.io/graalvm/native-image-community:21-ol9"
}
```

### Steps To Reproduce

see actual behaviour...

### Environment Information

_No response_

### Example Application

_No response_

### Version

4.2.3

### PR DESCRIPTION
**Pull request by timyates**
When using Java 21 for dockerimageNative we have to ensure we pass jdkVersion to get the right base image from the plugin. The dockerimage base image should also be adjusted accordingly.

Fixes #2281

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/build/MicronautBuildPlugin.java [hunk 1]
```diff
@@ -209,11 +223,19 @@ protected MicronautApplicationGradlePlugin.Builder micronautGradleApplicationPlu
         if (generatorContext.getFeatures().contains(AwsLambda.FEATURE_NAME_AWS_LAMBDA) && (
                 (generatorContext.getApplicationType() == ApplicationType.FUNCTION && generatorContext.getFeatures().contains(FEATURE_NAME_GRAALVM)) ||
                         (generatorContext.getApplicationType() == ApplicationType.DEFAULT))) {
-            builder = builder.dockerNative(Dockerfile.builder().baseImage("amazonlinux:2")
+            builder.dockerNative(Dockerfile.builder()
+                    .baseImage("amazonlinux:2")
```
Comment (melix) [inline starter-core/src/main/java/io/micronaut/starter/feature/build/MicronautBuildPlugin.java:227]: The default is 2023 in Gradle plugin 4.3.0, see https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/929

**Review by melix — APPROVED** [review]

**Review by sdelamo — CHANGES_REQUESTED** [review]
change the PR base to 4.3.x

File: starter-core/src/main/java/io/micronaut/starter/feature/build/MicronautBuildPlugin.java (hunk 1 above)
Comment (timyates) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/MicronautBuildPlugin.java:227]: This is 4.2.x

Comment (timyates) [reply starter-core/src/main/java/io/micronaut/starter/feature/build/MicronautBuildPlugin.java:227]: Looks like I need to rebase to 4.3.x

**Pull request closed without merging**